runtime/common/src/integration_tests.rs

```diff
@@ -864,39 +889,198 @@ fn basic_swap_works() {
 		// Withdraw of contributions works
 		assert_eq!(Balances::free_balance(&crowdloan_account), total);
 		for i in 10..20 {
-			assert_ok!(Crowdloan::withdraw(Origin::signed(i), i, ParaId::from(2001)));
+			assert_ok!(Crowdloan::withdraw(signed(i), account_id(i), ParaId::from(2001)));
 		}
 		assert_eq!(Balances::free_balance(&crowdloan_account), 0);
 
 		// Dissolve returns the balance of the person who put a deposit for crowdloan
-		assert_ok!(Crowdloan::dissolve(Origin::signed(1), ParaId::from(2001)));
-		assert_eq!(Balances::reserved_balance(&1), 0);
-		assert_eq!(Balances::reserved_balance(&2), 500 + 20 * 2 * 1);
+		assert_ok!(Crowdloan::dissolve(signed(1), ParaId::from(2001)));
+		assert_eq!(Balances::reserved_balance(&account_id(1)), 0);
+		assert_eq!(Balances::reserved_balance(&account_id(2)), 500 + 20 * 2 * 1);
 
 		// Final deregister sets everything back to the start
 		assert_ok!(Registrar::deregister(para_origin(2001).into(), ParaId::from(2001)));
-		assert_eq!(Balances::reserved_balance(&2), 0);
+		assert_eq!(Balances::reserved_balance(&account_id(2)), 0);
+	})
+}
```

Comment:
We have tests for: 
`swap(chain + crowdloan + lease, thread)`
`swap(chain + crowdloan + lease, chain + crowdloan + lease)`

If I follow correctly another likely scenario would be
`swap(chain + crowdloan + lease, thread + crowdloan)`